### PR TITLE
Add 'Show tab title' attribute to the Product Details block

### LINKED ISSUE #47895: Remove the tab description from the Product Details' front-end representation
The Product Details block displays a non-customizable large title of the currently selected tab on the front-end.

image

It isn't shown in the canvas.

image

Let's remove the header completely. If someone needs it, they can add it directly in the block editor or in each individual product's description.

### PR DESCRIPTION
**Pull request by Aljullu**
### Changes proposed in this Pull Request:

Closes #47895.

This PR adds a new _Show tab title in content_ attribute to the Product Details block, so the user can customize if the title will be displayed or not.

To accomplish this, we use two approaches:

* Filtering some filters so they return an empty array (`woocommerce_product_description_heading`, `woocommerce_product_additional_information_heading` and `woocommerce_reviews_title`).
* With `WP_HTML_Tag_Processor`, we hide the first `h2` inside `.wc-tab`.

### How to test the changes in this Pull Request:

1. With a block theme, go to Appearance > Editor > Templates > Single Product.
2. Convert the template to the blockified template if necessary, and select the Product Details block.
3. Verify that, by default, the title of each tab is displayed in the tab content.
image
4. Toggle the _Show tab title in content_ attribute. Verify the title disappears/appears as expected.

https://github.com/user-attachments/assets/3a316d82-3420-4c9a-af4b-53f1ef4e1103

5. Save and visit the product page on the frontend. Verify the tab title is displayed in content when the toggle is on, and it's not displayed when the toggle is off.

### CODE REVIEW

**Comment by Marc-pi** [comment]
ah good, that was strongly needed, i found the double title was strange :-)
thx

**Review by WunderBart — APPROVED** [review]
Followed the testing instructions – everything works great in both editor and front-end. Thanks! 🚀

File: plugins/woocommerce-blocks/tests/e2e/tests/single-product-details/single-product-details.block_theme.spec.ts [hunk 1]
```diff
@@ -55,4 +55,56 @@ test.describe( `${ blockData.slug } Block`, () => {
...
+			postType: 'wp_template',
+			canvas: 'edit',
+		} );
+
+		await editor.insertBlock( {
+			name: blockData.slug,
+		} );
+
+		const block = await editor.getBlockByName( blockData.slug );
+		await block.click();
```
Comment (WunderBart) [inline plugins/woocommerce-blocks/tests/e2e/tests/single-product-details/single-product-details.block_theme.spec.ts]: Can we use `editor.selectBlocks` here instead?

Comment (Aljullu) [reply plugins/woocommerce-blocks/tests/e2e/tests/single-product-details/single-product-details.block_theme.spec.ts]: Good point! Done in 3bc456b54f26e04456d7ce91262a755f584df9d2.

**Pull request merged**